src/ast/mod.rs

```diff
@@ -3968,6 +3968,19 @@ pub enum Statement {
         owned_by: Option<ObjectName>,
     },
     /// ```sql
+    /// CREATE DOMAIN name [ AS ] data_type
+    ///         [ COLLATE collation ]
+    ///         [ DEFAULT expression ]
+    ///         [ domain_constraint [ ... ] ]
+    ///
+    ///     where domain_constraint is:
+    ///
+    ///     [ CONSTRAINT constraint_name ]
+    ///     { NOT NULL | NULL | CHECK (expression) }
+    /// ```
```

Comment:
Instead of duplicating this doc comment, maybe it would suffice to only mention that this is a `CREATE DOMAIN` statement, [for example](https://github.com/apache/datafusion-sqlparser-rs/blob/d4868ba1316419ef3dc79942eaf1eaf662be359c/src/ast/mod.rs#L2980)

Reply:
Updated